### DISCUSSION TITLE
Any kind of intro/example project?

### DISCUSSION THREAD

**Discussion by skipsizemore** [open-telemetry/opentelemetry-go#3072] [General]
I've been doing Go full-time for 8 years, and this is EASILY the most frustrating open-source project I have ever tried to use. _All I want it to do_ is produce basic trace/span info and send it to an http endpoint. Trying to actually do this with the libraries available, their many conflicting module paths and versions, etc., is just maddening.

I can produce traces and dump them to stdout, but as soon as I try to send them anywhere, I am dead in the middle of dependency hell. It may be organized very cleverly for all I can tell, but without any docs that actually create a working app, it's hard to say. No example code I can find will compile.

Before I give up on this completely, is there a fully-working example project I can look at somewhere that does this extremely basic thing?

**Comment by Petrie** [comment]
Maybe this is what you need, a working example.
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/examples/demo

**Reply by srabraham** [reply]
I'd mention that that demo is out of date, since these two deps of it no longer exist:
go.opentelemetry.io/otel/sdk/metric/controller/basic
go.opentelemetry.io/otel/sdk/metric/processor/basic
see: https://github.com/open-telemetry/opentelemetry-go/tree/main/sdk/metric (no controller or processor subdirs anymore)

**Comment by dmathieu** [comment]
There is also this multi-apps demo: https://github.com/open-telemetry/opentelemetry-demo
